### DISCUSSION TITLE
Adding metaobject definitions under a reserved namespace

### DISCUSSION THREAD

**Discussion by AndrejMazhenkovski** [kinngh/shopify-node-mongodb-next-app#63] [General]
I was wondering where do we add the metafields that we want the app to be able to consume, and be reserved for our app only, in the setup? 

Here's a link for reference to creating that with the Remix template:  https://shopify.dev/docs/apps/build/checkout/cart-checkout-validation/create-complex-validation-rules/add-configuration 

Do we add the mutation inside of freshInstall.js, and the metafield definition inside shopify.js ? 

Appreciate your answer @kinngh!

**Comment by kinngh** [comment]
`freshInstall.js` and `shopify.js` files have nothing to do with it. It's just an API call to create the Metafield/object, just like you would with any other endpoint. If you want to create it as part of your onboarding / initial setup, then you throw it in `freshInstall.js` otherwise it's a regular API call to a GraphQL endpoint.

**Reply by AndrejMazhenkovski** [reply]
I guess the "definition" part threw me off, but looks like it's consumed by the GQL mutation and has nothing to do with shopify.js config. Thanks! 🙌
